datalad/runner/tests/test_nonasyncrunner.py

```diff
@@ -184,3 +330,204 @@ def test_popen_invocation(src_path, dest_path):
     fetching_data.start()
     fetching_data.join(5.0)
     assert_false(fetching_data.is_alive(), "Child is stuck!")
+
+
+def test_timeout():
+    # Expect timeout protocol calls on long running process
+    # if the specified timeout is short enough
+    class TestProtocol(WitlessProtocol):
+
+        received_timeouts = list()
+
+        def __init__(self):
+            WitlessProtocol.__init__(self)
+            self.counter = count()
+
+        def timeout(self, fd: Optional[int]):
+            TestProtocol.received_timeouts.append((self.counter.__next__(), fd))
+
+    result = run_command(
+        ['timeout', '3'] if on_windows else ["sleep", "5"],
+        stdin=None,
+        protocol=TestProtocol,
+        timeout=1
+    )
+    assert_true(len(TestProtocol.received_timeouts) > 0)
+    assert_true(all(map(lambda e: e[1] is None, TestProtocol.received_timeouts)))
+
+
+def test_timeout_nothing():
+    # Expect timeout protocol calls on long running process
+    # if the specified timeout is short enough
+    class TestProtocol(NoCapture):
+        def __init__(self,
+                     timeout_queue: List):
+            NoCapture.__init__(self)
+            self.timeout_queue = timeout_queue
+            self.counter = count()
+
+        def timeout(self, fd: Optional[int]) -> bool:
+            if fd is not None:
+                self.timeout_queue.append((self.counter.__next__(), fd))
+            return True
+
+    stdin_queue = queue.Queue()
+    for i in range(72):
+        stdin_queue.put(b"\x00" * 1024)
+    stdin_queue.put(None)
+
+    timeout_queue = []
+    run_command(
+        py2cmd("import time; time.sleep(5)\n"),
+        stdin=stdin_queue,
+        protocol=TestProtocol,
+        timeout=1,
```

Comment:
could
```suggestion
        timeout=0.1,
```
for the same effect?

Reply:
I changed `sleep(5)` in line 381 to `sleep(.6)` and added your change for line 384